[macOS] Display thumbnail of selected file for <input type=file>

rr-codes opened the pull request:
#### 43b96844e7e9fc0ceb95e46f46d0b515864d3fa2
<pre>
[macOS] Display thumbnail of selected file for &lt;input type=file&gt;
<a href="https://bugs.webkit.org/show_bug.cgi?id=245993">https://bugs.webkit.org/show_bug.cgi?id=245993</a>
rdar://100570349

Reviewed by Aditya Keerthi.

On iOS, we show a thumbnail of the selected file, but on macOS, we just
sure the generic file icon based on the file&apos;s extension.

This PR adopts the iOS behavior for macOS, and refactors out the thumbnail
decoration logic into `RenderThemeCocoa`. It also refactors out some of the
icon creation logic into common places for both iOS and macOS.

Adding a new file broke some unified build files so it also fixes those.

* Source/WebCore/platform/graphics/Icon.h:
(WebCore::Icon::image const):
(WebCore::Icon::nsImage const): Deleted.
* Source/WebCore/platform/graphics/mac/IconMac.mm:
(WebCore::Icon::Icon):
(WebCore::Icon::~Icon):
(WebCore::Icon::createIconForImage):
(WebCore::Icon::createIconForFiles):
(WebCore::Icon::createIconForFileExtension):
(WebCore::Icon::createIconForUTI):
(WebCore::Icon::paint):
(WebCore::squareCropRectForSize): Deleted.
(WebCore::squareImage): Deleted.
(WebCore::thumbnailSizedImageForImage): Deleted.
(WebCore::iconForImageFile): Deleted.
(WebCore::iconForVideoFile): Deleted.
(WebCore::iconForFile): Deleted.
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::iconForAttachment):
* Source/WebKit/Shared/Cocoa/WebIconUtilities.h: Copied from Source/WebKit/Shared/ios/WebIconUtilities.h.
* Source/WebKit/Shared/Cocoa/WebIconUtilities.mm: Renamed from Source/WebKit/Shared/ios/WebIconUtilities.mm.
(WebKit::squareCropRectForSize):
(WebKit::squareImage):
(WebKit::thumbnailSizedImageForImage):
(WebKit::fallbackIconForFile):
(WebKit::iconForImageFile):
(WebKit::iconForVideoFile):
(WebKit::iconForFiles):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStoreCollection.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStoreCollection.mm:
(WebKit::RemoteLayerWithRemoteRenderingBackingStoreCollection::allocateBufferForBackingStore):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
(-[_WKImageFileUploadItem displayImage]):
(-[_WKVideoFileUploadItem displayImage]):
(-[WKFileUploadPanel documentPicker:didPickDocumentsAtURLs:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandleSharing.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createIconForFiles): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClientCocoa.mm: Renamed from Source/WebKit/Shared/ios/WebIconUtilities.h.
(WebKit::WebChromeClient::createIconForFiles):
* Source/WebKit/WebProcess/WebCoreSupport/ios/WebChromeClientIOS.mm:
(WebKit::WebChromeClient::createIconForFiles): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebOpenPanelResultListener.mm:

Canonical link: <a href="https://commits.webkit.org/255355@main">https://commits.webkit.org/255355@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cb7f8d725d11b4063316fb924bd0da99d199bf5d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92082 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1314 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22595 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101875 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161930 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96082 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1311 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29712 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84524 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98051 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97740 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/823 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78607 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27764 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82745 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/82343 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70806 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/is-web-process-responsive (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36138 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16357 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33890 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17459 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3712 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37761 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40169 "Found 3 new test failures: http/wpt/service-workers/fetch-service-worker-preload.https.html, imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.html?h264_annexb, imported/w3c/web-platform-tests/webcodecs/videoDecoder-codec-specific.https.any.worker.html?h264_annexb (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39658 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36585 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebCore/platform/graphics/mac/IconMac.mm:
```diff
@@ -27,11 +27,141 @@
+
+    return image;
+
+//    return adoptNS([[NSWorkspace sharedWorkspace] iconForFileType:[@"." stringByAppendingString:fileExtension]]);
```
Please delete this line

hortont424 commented on Source/WebCore/platform/graphics/mac/IconMac.mm:
```diff
@@ -27,11 +27,141 @@
+
+    if (UTTypeConformsTo(fileUTI.get(), kUTTypeMovie))
+        return iconForVideoFile(file);
+ALLOW_DEPRECATED_DECLARATIONS_END
```
Please figure out which indentation is correct for these, and use it everywhere. Also probably merge this one with the following one.

hortont424 commented on Source/WebCore/rendering/RenderThemeCocoa.h:
```diff
@@ -25,22 +25,73 @@
+const float ControlBaseHeight = 20;
+const float ControlBaseFontSize = 11;
+
+struct IOSGradient {
```
Definitely should not be "IOS"-prefixed anymore.

hortont424 commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -78,11 +169,133 @@ @implementation WebCoreRenderThemeBundle
+        paintInfo.context().fillRect(thumbnailRect, backgroundImageColor);
+        {
+            GraphicsContextStateSaver stateSaver2(paintInfo.context());
+            CGContextRef cgContext = paintInfo.context().platformContext();
```
I'm confused. How is this OK with DOMGPUP on?

rr-codes replied:
what do you mean?

hortont424 replied:
I mean that "calling platformContext() and using CG in the Web process is usually an insta-crash with DOMGPUP on, so how is this working?"

rr-codes replied:
is DOMGPUP enabled by default? Because otherwise I have no idea how it's working 😅

hortont424 replied:
It is on iOS, yes.

rr-codes replied:
ya I have no idea then, since it works fine on both iOS and macOS 😅 I'll try testing again

rr-codes replied:
I just tested it again, and it works. And Also none of this code is new for iOS

hortont424 commented on Source/WebCore/rendering/RenderThemeCocoa.h:
```diff
@@ -25,22 +25,73 @@
+    static CGFunctionRef getSharedFunctionRef(IOSGradientRef, Interpolation);
+
+    static void drawAxialGradient(CGContextRef, IOSGradientRef, const FloatPoint&, const FloatPoint&, Interpolation);
+
```
This paragraphing is pretty loose.

rr-codes replied:
as In I should remove the newlines between each function?

hortont424 replied:
Or put them in sensible groups

pxlcoder commented on Source/WebCore/platform/graphics/mac/IconMac.mm:
```diff
@@ -27,24 +27,34 @@
+    ASSERT(m_cgImage);
 }
 
+
```
Nit: Unnecessary newline.

pxlcoder commented on Source/WebCore/platform/graphics/mac/IconMac.mm:
```diff
@@ -63,13 +73,17 @@
 
-        return adoptRef(new Icon(image));
+        PlatformImagePtr platformImage = [image CGImageForProposedRect:nil context:nil hints:nil];
+
```
I would remove the newline here, and the similar 3 cases below.

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -62,6 +61,11 @@ @implementation WebCoreRenderThemeBundle
+const int kThumbnailBorderStrokeWidth = 1;
+const int kThumbnailBorderCornerRadius = 1;
+const int kVisibleBackgroundImageWidth = 1;
+const int kMultipleThumbnailShrinkSize = 2;
```
`constexpr`

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -83,6 +87,45 @@ @implementation WebCoreRenderThemeBundle
+
+    IntSize cornerSize(kThumbnailBorderCornerRadius, kThumbnailBorderCornerRadius);
+
+    Color pictureFrameColor = this->pictureFrameColor(buttonRenderer);
```
`auto`

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -83,6 +87,45 @@ @implementation WebCoreRenderThemeBundle
+        thumbnailRect.contract(kMultipleThumbnailShrinkSize, kMultipleThumbnailShrinkSize);
+
+        // Background picture frame and simple background icon with a gradient matching the button.
+        Color backgroundImageColor = buttonRenderer.style().visitedDependentColor(CSSPropertyBackgroundColor);
```
`auto`

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -83,6 +87,45 @@ @implementation WebCoreRenderThemeBundle
+    Color pictureFrameColor = this->pictureFrameColor(buttonRenderer);
+
+    IntRect thumbnailPictureFrameRect = rect;
+    IntRect thumbnailRect = rect;
```
`auto`

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -83,6 +87,45 @@ @implementation WebCoreRenderThemeBundle
+
+    Color pictureFrameColor = this->pictureFrameColor(buttonRenderer);
+
+    IntRect thumbnailPictureFrameRect = rect;
```
`auto`

pxlcoder commented on Source/WebCore/rendering/RenderThemeMac.mm:
```diff
@@ -2593,7 +2593,8 @@ static Color titleTextColorForAttachment(const RenderAttachment& attachment, Att
 
     if (auto icon = WebCore::iconForAttachment(fileName, attachmentType, title))
-        return icon->nsImage();
+        return [[NSImage alloc] initWithCGImage:icon->image()->platformImage().get() size:NSZeroSize];
```
Needs an `adoptNS`.

pxlcoder commented on Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
```diff
@@ -146,7 +146,7 @@ - (BOOL)isVideo
 - (RetainPtr<UIImage>)displayImage
 {
-    return WebKit::iconForImageFile(self.fileURL);
+    return [[UIImage alloc] initWithCGImage:iconForImageFile(self.fileURL).get()];
```
`adoptNS`

pxlcoder commented on Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
```diff
@@ -164,7 +164,7 @@ - (BOOL)isVideo
 - (RetainPtr<UIImage>)displayImage
 {
-    return WebKit::iconForVideoFile(self.fileURL);
+    return [[UIImage alloc] initWithCGImage:iconForVideoFile(self.fileURL).get()];
```
`adoptNS`

pxlcoder commented on Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
```diff
@@ -891,7 +891,8 @@ - (void)documentPicker:(UIDocumentPickerViewController *)controller didPickDocum
         [retainedSelf->_view _removeTemporaryDirectoriesWhenDeallocated:WTFMove(temporaryURLs)];
         RunLoop::main().dispatch([retainedSelf = WTFMove(retainedSelf), maybeMovedURLs = WTFMove(maybeMovedURLs)] {
-            [retainedSelf _chooseFiles:maybeMovedURLs.get() displayString:displayStringForDocumentsAtURLs(maybeMovedURLs.get()) iconImage:WebKit::iconForFile(maybeMovedURLs.get()[0]).get()];
+            UIImage *icon = [[UIImage alloc] initWithCGImage:iconForFiles({ maybeMovedURLs.get()[0].absoluteString }).get()];
```
`adoptNS`

pxlcoder commented on Source/WebCore/platform/graphics/mac/IconMac.mm:
```diff
@@ -27,24 +27,34 @@
+        return nullptr;
+
+    return adoptRef(new Icon(NativeImage::create(WTFMove(image))));
+}
```
This looks like the same code in `IconIOS` – can we make an `IconCocoa`?

pxlcoder commented on Source/WebCore/rendering/RenderThemeIOS.mm:
```diff
@@ -130,7 +129,6 @@ static void interpolateLinearGradient(void *info, const CGFloat *inData, CGFloat
     IOSGradientRef gradient = static_cast<IOSGradientRef>(info);
     float alpha = inData[0];
     float inverse = 1.0f - alpha;
-
```
This change is unnecessary. (+ 3 others below)

pxlcoder commented on Source/WebCore/rendering/RenderThemeMac.mm:
```diff
@@ -2593,7 +2593,8 @@ static Color titleTextColorForAttachment(const RenderAttachment& attachment, Att
 
     if (auto icon = WebCore::iconForAttachment(fileName, attachmentType, title))
-        return icon->nsImage();
+        return adoptNS([[NSImage alloc] initWithCGImage:icon->image()->platformImage().get() size:NSZeroSize]);
```
Is `NSZeroSize` the right thing to use here?

rr-codes replied:
yes; it means "use the size of the cg image"

pxlcoder commented on Source/WebCore/SourcesCocoa.txt:
```diff
@@ -406,6 +406,7 @@ platform/graphics/cocoa/FontPlatformDataCocoa.mm
 platform/graphics/cocoa/GraphicsContextCocoa.mm
 platform/graphics/cocoa/GraphicsContextGLIOSurfaceSwapChain.cpp
 platform/graphics/cocoa/HEVCUtilitiesCocoa.mm
+platform/graphics/cocoa/IconCocoa.mm
```
Sort order: this should be after `platform/graphics/cocoa/IOSurfacePoolCocoa.mm`.

pxlcoder commented on Source/WebCore/platform/graphics/Icon.h:
```diff
@@ -26,12 +26,13 @@
-#elif PLATFORM(WIN)
+#endif
+
+#if PLATFORM(WIN)
```
Can this be kept as an `#elif`?

pxlcoder commented on Source/WebKitLegacy/mac/WebCoreSupport/WebOpenPanelResultListener.mm:
```diff
@@ -31,6 +31,7 @@
 
 #if PLATFORM(IOS_FAMILY)
 #import <WebCore/Icon.h>
+#import <WebCore/NativeImage.h>
```
Is this import necessary? I don't see any uses of `NativeImage` in this file.

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -133,8 +163,17 @@ static CGRect squareCropRectForSize(CGSize size)
+    // files, we only use the first URL to generate an icon.
+    NSURL *file = [NSURL fileURLWithPath:filenames[0] isDirectory:NO];
+    if (!file)
+        return nullptr;
```
This should be `return nil;`.

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -27,13 +27,21 @@
 
 #if PLATFORM(IOS_FAMILY)
-
 #import "UIKitSPI.h"
```
This import should be moved underneath the `#if PLATFORM(COCOA)` since the rest of the file is wrapped in there.

But, do we even need this import? I don't see any obvious uses of SPI here. Let's remove it if possible.

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.mm:
```diff
@@ -83,6 +87,45 @@ @implementation WebCoreRenderThemeBundle
+        paintInfo.context().fillRect(thumbnailRect, backgroundImageColor);
+
+        // Move the rects for the Foreground picture frame and icon.
+        int inset = kVisibleBackgroundImageWidth + kThumbnailBorderStrokeWidth;
```
`auto`

pxlcoder commented on Source/WebCore/rendering/RenderThemeCocoa.h:
```diff
@@ -25,7 +25,11 @@
+#include "GraphicsContextCG.h"
+#include "Icon.h"
 #include "RenderTheme.h"
+#include <CoreGraphics/CoreGraphics.h>
```
Are any of the imports actually needed in this header? We should import in the implementation files when possible.

pxlcoder commented on Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandleSharing.h:
```diff
@@ -26,6 +26,7 @@
 #pragma once
 
 #include "ImageBufferBackendHandle.h"
+#include "ImageBufferBackendHandleSharing.h"
```
Isn't this the file you're in right now?

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +65,68 @@ static CGRect squareCropRectForSize(CGSize size)
+    CGImageRef scaledImage = CGBitmapContextCreateImage(context);
+    if (!scaledImage) {
+        CGContextRelease(context);
+        return squaredImage;
```
`adoptCF`

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +65,68 @@ static CGRect squareCropRectForSize(CGSize size)
+
+    CGContextRelease(context);
+
+    return scaledImage;
```
`adoptCF`

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
+static PlatformImagePtr thumbnailSizedImageForImage(CGImageRef image)
 {
-    UIImage *squaredImage = squareImage(image);
+    CGImageRef squaredImage = squareImage(image).get();
```
Actually, rather than use `adoptCF` in the return statements, we can do:

`auto squaredImage = adoptCF(squareImage(image));` here.

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
+    CGContextSetInterpolationQuality(context, kCGInterpolationHigh);
+    CGContextDrawImage(context, destinationRect, squaredImage);
+
+    CGImageRef scaledImage = CGBitmapContextCreateImage(context);
```
`auto scaledImage = adoptCF(CGBitmapContextCreateImage(context));`

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
+    CGImageRef scaledImage = CGBitmapContextCreateImage(context);
+    if (!scaledImage) {
+        CGContextRelease(context);
+        return adoptCF(squaredImage);
```
`return squaredImage;`

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
+
+    CGContextRelease(context);
+
+    return adoptCF(scaledImage);
```
`return scaledImage;`

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
+    CFRelease(colorSpace);
+
+    CGContextSetInterpolationQuality(context, kCGInterpolationHigh);
+    CGContextDrawImage(context, destinationRect, squaredImage);
```
`squaredImage.get()`

pxlcoder commented on Source/WebKit/Shared/Cocoa/WebIconUtilities.mm:
```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
+    CGRect destinationRect = CGRectMake(0, 0, iconSideLength, iconSideLength);
+
+    CGColorSpaceRef colorSpace = CGImageGetColorSpace(image);
+    CFRetain(colorSpace);
```
You can remove all the `CFRetain`/`CFRelease` and use `RetainPtr`/`adoptCF` instead.

rr-codes committed:
[macOS] Display thumbnail of selected file for <input type=file>
https://bugs.webkit.org/show_bug.cgi?id=245993
rdar://100570349

Reviewed by Aditya Keerthi.

On iOS, we show a thumbnail of the selected file, but on macOS, we just
sure the generic file icon based on the file's extension.

This PR adopts the iOS behavior for macOS, and refactors out the thumbnail
decoration logic into `RenderThemeCocoa`. It also refactors out some of the
icon creation logic into common places for both iOS and macOS.

Adding a new file broke some unified build files so it also fixes those.

* Source/WebCore/platform/graphics/Icon.h:
(WebCore::Icon::image const):
(WebCore::Icon::nsImage const): Deleted.
* Source/WebCore/platform/graphics/mac/IconMac.mm:
(WebCore::Icon::Icon):
(WebCore::Icon::~Icon):
(WebCore::Icon::createIconForImage):
(WebCore::Icon::createIconForFiles):
(WebCore::Icon::createIconForFileExtension):
(WebCore::Icon::createIconForUTI):
(WebCore::Icon::paint):
(WebCore::squareCropRectForSize): Deleted.
(WebCore::squareImage): Deleted.
(WebCore::thumbnailSizedImageForImage): Deleted.
(WebCore::iconForImageFile): Deleted.
(WebCore::iconForVideoFile): Deleted.
(WebCore::iconForFile): Deleted.
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::iconForAttachment):
* Source/WebKit/Shared/Cocoa/WebIconUtilities.h: Copied from Source/WebKit/Shared/ios/WebIconUtilities.h.
* Source/WebKit/Shared/Cocoa/WebIconUtilities.mm: Renamed from Source/WebKit/Shared/ios/WebIconUtilities.mm.
(WebKit::squareCropRectForSize):
(WebKit::squareImage):
(WebKit::thumbnailSizedImageForImage):
(WebKit::fallbackIconForFile):
(WebKit::iconForImageFile):
(WebKit::iconForVideoFile):
(WebKit::iconForFiles):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStoreCollection.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerWithRemoteRenderingBackingStoreCollection.mm:
(WebKit::RemoteLayerWithRemoteRenderingBackingStoreCollection::allocateBufferForBackingStore):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/ios/forms/WKFileUploadPanel.mm:
(-[_WKImageFileUploadItem displayImage]):
(-[_WKVideoFileUploadItem displayImage]):
(-[WKFileUploadPanel documentPicker:didPickDocumentsAtURLs:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandleSharing.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createIconForFiles): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClientCocoa.mm: Renamed from Source/WebKit/Shared/ios/WebIconUtilities.h.
(WebKit::WebChromeClient::createIconForFiles):
* Source/WebKit/WebProcess/WebCoreSupport/ios/WebChromeClientIOS.mm:
(WebKit::WebChromeClient::createIconForFiles): Deleted.
* Source/WebKitLegacy/mac/WebCoreSupport/WebOpenPanelResultListener.mm:

Canonical link: https://commits.webkit.org/255355@main
